cocos/3d/CCSprite3D.cpp

```diff
@@ -465,22 +466,21 @@ void Sprite3D::genGLProgramState(bool useLight)
 {
     _shaderUsingLight = useLight;
 
-    std::unordered_map<const MeshVertexData*, GLProgramState*> glProgramestates;
+    std::unordered_map<const MeshVertexData*, Sprite3DMaterial*> materials;
```

Comment:
I would prefer using `genMaterial` function instead of the old one if it does not break too much compatibility.